### PR TITLE
Retry for ProvisioningRequests KEP

### PR DESCRIPTION
**Pull request by mimowo**
#### What type of PR is this?

/kind documentation
/kind feature

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes-sigs/kueue/issues/1353

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by mimowo** [comment]
cc @mwielgus @alculquicondor

File: keps/1136-provisioning-request-support/README.md [hunk 1]
```diff
@@ -146,6 +153,26 @@ type ProvisioningRequestConfigSpec struct {
...
+	//
+	// +optional
+	// +kubebuilder:default=3
+	MaxRetries *int32 `json:"maxRetries,omitempty"`
+
+	// DefaultBackoffSeconds indicates the default backoff in seconds. The intervals
+	// between consecutive attempts are determined by expotential growth up to 30min.
+	// +optional
+	// +kubebuilder:default=60
+	DefaultBackoffSeconds *int32 `json:"minBackoffSeconds,omitempty"`
```
[the remark this answers is not in the document]
Comment (mimowo) [reply keps/1136-provisioning-request-support/README.md:175]: Yeah, I was hesitant, `minBackoffSeconds` sgtm

**Comment by mimowo** [comment]
@alculquicondor should we move forward with the KEP note, or close it?

**Comment by alculquicondor** [comment]
I think there is value in making it configurable in future versions, but I'm ok deferring this for later, based on user feedback.

**Comment by mimowo** [comment]
> I think there is value in making it configurable in future versions, but I'm ok deferring this for later, based on user feedback.

Sure, for now the PR is just a note about the retry itself. Does not describe the API for configuration. Wondering if the note itself is worth it.

File: keps/1136-provisioning-request-support/README.md [hunk 2]
```diff
@@ -105,6 +105,11 @@ If the `ProvisioningRequest` fails, fail the `AdmissionCheck`.
 the provisioning request should also be deleted (the last one can be achieved via
 OwnerReference).
 
+* Retry ProvisioningRequests with respect to the `RetryConfig` configuration in
+the `ProvisioningRequestConfig`. For each attempt a new provisioning request is
+created with the suffix indicating the attempt number. The corresponding admission
+check will remain in the `Pending` state until the retries end.
```
Comment (alculquicondor) [inline keps/1136-provisioning-request-support/README.md:111]: Add the note about the number of retries

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**